### PR TITLE
New rule `keyframe-declaration-no-important`.

### PR DESCRIPTION
**Pull request by evilebottnawi**
/cc @davidtheclark

### CODE REVIEW

**Comment by jeddy3** [comment]
Wow, you two have been busy! I'll take a closer look at all the changes when I'm back on Monday. From the cursory look I've taken, everything looks fab!. Just thought I'd quickly comment on this PR though (while I have WiFi on my phone), to say I think this rule should be called `keyframe-declaration-no-important`. `at-rule-*` rules target all at-rules, and aren't geared towards targeting at-rules of a specific type. Keyframe is a valid thing, being shorthand for "keyframe style rule" (from the spec). It should belong it's section (called "Keyframe rule"), and should come just before "Root rule". Does that sound good? P.S. I think this is a great rule and should be included in the standard config :)

**Comment by davidtheclark** [comment]
Yep, sounds great. I'll change the name.

**Comment by evilebottnawi** [comment]
@jeddy3 Thanks, rename, we wait you :smile:

File: docs/user-guide/rules.md [hunk 1]
```diff
@@ -213,6 +213,10 @@ Here are all the rules within stylelint, grouped by the [*thing*](http://apps.wo
 - [`comment-whitespace-inside`](../../src/rules/comment-whitespace-inside/README.md): Require or disallow whitespace on the inside of comment markers.
 - [`comment-word-blacklist`](../../src/rules/comment-word-blacklist/README.md): Specify a blacklist of disallowed words within comments.
 
+### Keyframe rule
+
+- [`keyframe-declaration-no-important`](../../src/rules/keyframe-declaration-no-important/README.md): Disallow `!important` within declarations in a keyframe.
```
Comment (davidtheclark) [inline docs/user-guide/rules.md]: a keyframe rule.

File: src/rules/index.js [hunk 2]
```diff
@@ -202,6 +203,7 @@ export default {
   "function-whitelist": functionWhitelist,
   "function-whitespace-after": functionWhitespaceAfter,
   "indentation": indentation, // eslint-disable-line object-shorthand
+  "keyframe-declaration-no-important": keyframeDeclarationNoImportant,
```
Comment (davidtheclark) [inline src/rules/index.js]: Are these extra changes just alphabetization? Could you double-check them?

File: src/rules/keyframe-declaration-no-important/README.md [hunk 3]
```diff
@@ -0,0 +1,64 @@
+# keyframe-declaration-no-important
+
+Disallow `!important` within declarations in a keyframe.
```
Comment (davidtheclark) [inline src/rules/keyframe-declaration-no-important/README.md]: keyframe rule.

File: src/rules/keyframe-declaration-no-important/index.js [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+export const messages = ruleMessages(ruleName, {
+  rejected: "Unexpected !important within declarations in a keyframe",
+})
+
+export default function (actual) {
+  return (root, result) => {
+    const validOptions = validateOptions(result, ruleName, { actual })
+    if (!validOptions) { return }
+
+    root.walkAtRules(/^keyframes$/i, atRuleKeyframes => {
```
Comment (davidtheclark) [inline src/rules/keyframe-declaration-no-important/index.js]: Nice job remembering the case insensitivity!

**Comment by davidtheclark** [comment]
LGTM, except a couple of words to add, I think.

File: src/rules/index.js (hunk 2 above)
Comment (evilebottnawi) [inline src/rules/index.js]: `keyframe-declaration-no-important` go after `indentation` in alphabet order.

**Pull request merged**